drivers/entropy/Kconfig

```diff
@@ -5,7 +5,7 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 
 config ENTROPY_CC3XX
-	bool "Arm CC3XX RNG driver for Nordic devices" if !BUILD_WITH_TFM
+	bool "Arm CC3XX RNG driver for Nordic devices"
```

Comment:
Reading the commit message and looking at this code.
> Allow the CC3XX entropy driver to be disabled with TF-M when the
> Crypto partition is disabled.
> This would otherwise cause a compilation error.

then i'm not sure I fully understand.

Now the cc3xx entropy driver can always be enabled, also when `TFM_PARTITION_CRYPTO=y`.
Based on the description in the commit message and this code change, I assume the following is now possible but will most likely result in a compilation error:
```
cmake ... -DCONFIG_BUILD_WITH_TFM=y -DCONFIG_TFM_PARTITION_CRYPTO=y -DCONFIG_ENTROPY_CC3XX=y
```

I don't believe this is intended.
Could you please check this, and update code or commit message if needed.

Reply:
right. seems I overlooked that code change, should be fine then.

Note, generally speaking, having a select directly to a specific driver can often be problematic, as it prevents alternative driver implementations, though in this particular case it is acceptable.

Reply:
@joerchan still I do not understand this part of the commit message ?
> Allow the CC3XX entropy driver to be disabled with TF-M when the Crypto partition is disabled.
> This would otherwise cause a compilation error.

Is it allowed to:
- Enable TF-M
- Disable tfm partition crypto
- Enable entropy cc3xx

Or what / when is the compilation error you refer to seen ?
I would like to understand the issue you are trying to fix, and thus be able to reproduce this issue and then see this PR actually solves this problem.

Reply:
That would lead to a compilation error. The same one that I wanted to get rid of. 
```
entropy_cc310.c.obj): in function `entropy_cc3xx_rng_get_entropy':
entropy_cc310.c:42: undefined reference to `psa_generate_random'
```

The problem was really that ENTROPY_CC3XX was always enabled when TF-M is enabled. But the psa_generate_random is only available when TFM crypto partition with RNG support is enabled.

SPM also have the same problem but with SPM_SERVICE_RNG as a dependent configuration.

Having the select from SPM / TFM seems wrong to me now. Since SPM / TFM does not depend on this.
Instead the ENTROPY driver can use these services when they are enabled.

Reply:
> @tejlmand Trying to fix the above leads me down the rabbit hole of additional changes.

I can imagine.

> The commit itself is not needed to fix the TFM minimal configuration, so I have taken it out of this PR and will do a follow up PR later once I've figured out the rest.

Also note, that I will gladly approve even a minor improvement of minimizing the risk of getting compilation errors.
Just trying to ensure that if a commit is stating as fixing a particular issue, then it actually does fix it.
Otherwise it should just stated that it improves user-experience, cause then it's easier to decide which use-case is most important.

Reply:
That's ok. The case I was trying to fix is really hard to get to, so not something that we have to address at the moment.